pkg/process/dockerproxy/filter.go

```diff
@@ -0,0 +1,164 @@
+// +build !windows
+
+package dockerproxy
+
+import (
+	"strconv"
+	"strings"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/gopsutil/process"
+)
+
+// Filter keeps track of every docker-proxy instance and filters network traffic going through them
+type Filter struct {
+	proxyByTarget map[target]*proxy
+	// This "secondary index" is used only during the proxy IP discovery process
+	proxyByPID map[int32]*proxy
+}
+
+type target struct {
+	ip    string
+	port  int32
+	proto model.ConnectionType
+}
+
+type proxy struct {
+	pid    int32
+	ip     string
+	target target
+}
+
+// NewFilter instantiates a new filter loaded with docker-proxy instance information
+func NewFilter() *Filter {
+	filter := new(Filter)
+	if procs, err := process.AllProcesses(); err == nil {
```

Comment:
Isn't calling `AllProcesses` fairly expensive?

Reply:
This looks like it's only getting called once, which isn't too bad (we usually run it every 10s, or 2s for RT).

Reply:
I think there are race conditions here:

- What if the `docker-proxy` process restarts?
- What if the `process-agent`  process comes up before the `docker-proxy` process?

Reply:
> This looks like it's only getting called once, which isn't too bad

It looks like it gets called once per connection check

Reply:
Aha - we're creating a new filter every time. Good catch - disregard my comment. That does mean it'll still happen every 30s, which may be fine.